evalml/tests/model_understanding_tests/test_partial_dependence.py

```diff
@@ -3008,3 +3008,123 @@ def test_partial_dependence_dfs_transformer_does_not_calculate_feature_matrix(
     assert part_dep.feature_values.notnull().all()
     assert part_dep.partial_dependence.notnull().all()
     pd.testing.assert_frame_equal(part_dep, fast_part_dep)
+
+
+@pytest.mark.parametrize(
+    "nullable_y_ltype",
+    ["IntegerNullable", "AgeNullable", "BooleanNullable"],
+)
+@pytest.mark.parametrize(
+    "grid_resolution",
+    [2, 10, 50],
+)
+@pytest.mark.parametrize(
+    "use_int_null_many_values",
+    [True, False],
+)
+def test_partial_dependence_with_nullable_types(
+    nullable_type_test_data,
+    nullable_type_target,
+    linear_regression_pipeline,
+    logistic_regression_binary_pipeline,
+    nullable_y_ltype,
+    grid_resolution,
+    use_int_null_many_values,
+):
+    y = nullable_type_target(ltype=nullable_y_ltype, has_nans=False)
+    X = nullable_type_test_data(has_nans=False)
+    X = X.ww.select(["numeric", "Boolean", "BooleanNullable"])
+
+    int_col = "int col nullable"
+    if use_int_null_many_values:
+        X.ww[int_col] = ww.init_series(
+            pd.Series(range(len(y))),
+            logical_type="IntegerNullable",
+        )
+
+    pipeline = linear_regression_pipeline
+    if nullable_y_ltype == "BooleanNullable":
+        pipeline = logistic_regression_binary_pipeline
```

Comment:
What's the reasoning behind this?

Reply:
I want to be able to test partial dependence with various y logical types, so setting different pipelines here lets me test boolean nullable (the linear regression pipeline errors with `Regression pipeline can only handle numeric target data` otherwise)